benchmarks/optcarrot/tools/plot.rb

```diff
@@ -9,58 +9,116 @@
 mpl.use("agg")
 
 pyimport "matplotlib.pyplot", as: "plt"
+pyimport "matplotlib.patches", as: "patches"
+pyimport "matplotlib.path", as: "path"
 
-[true, false].each do |oneshot|
-  df = pd.read_csv(oneshot ? ARGV[0] : ARGV[1], index_col: ["mode", "name"])
+if ARGV.size < 2
+  puts "Usage: #$0 benchmark/...oneshot-180.csv benchmark/...oneshot-3000.csv"
+end
+
+[180, 3000].each do |frames|
+  df = pd.read_csv(frames == 180 ? ARGV[0] : ARGV[1], index_col: ["mode", "name"])
   df = df.filter(regex: "run \\d+").stack().to_frame("fps")
   idx = df.index.drop_duplicates
   gp = df["fps"].groupby(level: ["mode", "name"])
-  (oneshot ? [true, false] : [true]).each do |summary|
+  [true, false].each do |summary|
     mean, std = [gp.mean(), gp.std()].map do |df_|
       df_ = df_.unstack("mode")
       df_ = df_.reindex(index: idx.get_level_values("name").unique)
       df_ = df_.reindex(columns: idx.get_level_values("mode").unique)
-      df_ = df_["default"].fillna(df_["opt-none"]).to_frame if oneshot && summary
+      df_ = df_["default"].fillna(df_["opt-none"]).to_frame if summary
       df_
     end
-    ax = mean.plot(
-      kind: :barh, figsize: [8, oneshot ? summary ? 7 : 13 : 2], width: 0.8,
-      xerr: std, ecolor: "lightgray", legend: !summary)
-    ax.legend(title: "mode", loc: "lower right") unless summary
-    ax.set_title(
-      oneshot ?
-        "Ruby implementation benchmark with Optcarrot (180 frames)"
-      :
-        "Start-up time (the time to show the first frame)"
+
+    d = mean + std
+    break_start = max = d.max.max.to_f.ceil(-1) + 10
+    if frames == 3000
+      d = mean + std
+      break_start = d[d.index != "truffleruby"].max.max.to_f.ceil(-1) + 10
+      d = mean - std
+      break_end = d[d.index == "truffleruby"].min.min.to_f.floor(-1) - 10
```

Comment:
What does this do?

Reply:
Looks like this is its commit of origin in optcarrot: https://github.com/mame/optcarrot/commit/765d9ed6b4916fe43dfd44d994ed253a33e220a3

Not 100% sure, but I know some of their graphs break out TruffleRuby separately since its scale messes with the axis height. This looks like it may be what does that.

Reply:
I'm guessing this is the output graph that the Truffle-specific break is mean to be part of:

https://github.com/mame/optcarrot/blob/master/doc/benchmark-summary-3000.png

Reply:
Yes. I imported the latest Optcarrot master.
The actual change regarding Fibers is https://github.com/mame/optcarrot/pull/35